Dashboard/app/js/lib/views/data/question-answer-view.js

```diff
@@ -157,6 +161,81 @@ FLOW.QuestionAnswerView = Ember.View.extend({
     return null;
   }.property('this.content'),
 
+  /*
+  * Parse the caddisfly test JSON result
+  * Extracts the 'result' key from a json string, and parses the array of results.
+  * Creates an array with name, value, unit for each result
+  * Example JSON format: {"result":[{"name":"Total Chlorine (ppm)","value":10,
+  * "unit":"ppm","id":0},{"name":"Free Chlorine (ppm)","value":0.5,"unit":"ppm",
+  * "id":1}],"type":"caddisfly","name":"Chlorine and Free Chlorine",
+  * "uuid":"bf1c19c0-9788-4e26-999e-1b5c6ca28111","image":"b3893f16-6a02-4e92-a13e-fce25223a0c5.png"}
+  */
+  parseTestJson: function(){
+	  var c = this.content, testJson, newResult, image;
+  	result=Ember.A();
+  	if (c && c.get('value')) {
+  	  testJson = JSON.parse(c.get('value'));
+  	  if (testJson.result && !Ember.empty(testJson.result)){
+ 		    testJson.result.forEach(function(item){
+  			  newResult = {
+  			    "name":item.name,
+  				  "value":item.value,
+  				  "unit":item.unit,
+  	  		};
+  		  	result.push(newResult);
+  		  });
+  	  }
+    }
+    this.set('testResult',result);
```

Comment:
I don't think so, because there can be multiple results, so the array will be filled with objects, each having a name, value and unit.